lib/rspec/expectations/fail_with.rb

```diff
@@ -24,8 +24,7 @@ def fail_with(message, expected=nil, actual=nil)
                                "appropriate failure_message[_when_negated] method to return a string?"
         end
 
-        diff = differ.diff(actual, expected)
-        message = "#{message}\nDiff:#{diff}" unless diff.empty?
+        message = ::RSpec::Matchers::ExpectedsForMultipleDiffs.from(expected).message_with_diff(message, method(:differ), actual)
```

Comment:
Yes, now I see. For example when you do a `diff` on the same `differ` twice you get:

```
       Diff for (include "bar"):
       @@ -1,2 +1,3 @@
       -bar
       +baz
       +bug
       
       Diff for (include "foo"):
       @@ -1,2 +1,3 @@
       --bar
       +baz
       +bu
```